src/hooks/useUsersGuildPins.ts

```diff
@@ -8,7 +8,89 @@ import {
   GUILD_PIN_CONTRACTS,
   GuildPinsSupportedChain,
 } from "utils/guildCheckout/constants"
-import { createPublicClient, http } from "viem"
+import { PublicClient, createPublicClient, http } from "viem"
+
+const getUsersGuildPinIdsOnChain = async (
+  balance: bigint,
+  chain: GuildPinsSupportedChain,
+  address: `0x${string}`,
+  client: PublicClient
+) => {
+  const contracts = Array.from({ length: Number(balance) }, (_, i) => ({
+    abi: GUILD_PIN_CONTRACTS[chain].abi,
+    address: GUILD_PIN_CONTRACTS[chain].address,
+    functionName: "tokenOfOwnerByIndex",
+    args: [address, BigInt(i)],
+  }))
+
+  const results =
+    contracts.length > 0
+      ? await client.multicall({
+          contracts: contracts,
+        })
+      : []
+
+  const errors: Error[] = results
+    .filter((result) => result.status === "failure")
+    .map((result) => result.error)
+
+  const pinIds: bigint[] = results
+    .filter((result) => result.status === "success")
+    .map((result) => result.result as bigint)
+
+  return { pinIds, errors }
+}
+
+const getPinTokenURIsForPinIds = async (
+  pinIds: bigint[],
+  chain: GuildPinsSupportedChain,
+  client: PublicClient
+) => {
+  const contractCalls = pinIds.map((tokenId) => ({
+    abi: GUILD_PIN_CONTRACTS[chain].abi,
+    address: GUILD_PIN_CONTRACTS[chain].address,
+    functionName: "tokenURI",
+    args: [tokenId],
+  }))
+
+  const results =
+    contractCalls.length > 0
+      ? await client.multicall({
+          contracts: contractCalls,
+        })
+      : []
+
+  const tokenURIs = results.map((result, index) => ({
```

Comment:
This variable name is a bit misleading, `tokenURI` is a base64 encoded metadata, this variable contains that base64 string, the chainId, and the tokenId too, so let's find a better name for it please